core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -490,15 +498,19 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         totalTokensOutstanding = totalTokensOutstanding.sub(tokensToRemove);
     }
 
-    function _deleteSponsorPosition(address sponsor) internal {
+    function _deleteSponsorPosition(address sponsor) internal returns (FixedPoint.Unsigned memory) {
         PositionData storage positionToLiquidate = _getPositionData(sponsor);
 
         // Remove the collateral and outstanding from the overall total position.
-        rawTotalPositionCollateral = rawTotalPositionCollateral.sub(positionToLiquidate.rawCollateral);
+        FixedPoint.Unsigned memory remainingRawCollateral = positionToLiquidate.rawCollateral;
+        rawTotalPositionCollateral = rawTotalPositionCollateral.sub(remainingRawCollateral);
         totalTokensOutstanding = totalTokensOutstanding.sub(positionToLiquidate.tokensOutstanding);
 
         // Reset the sponsors position to have zero outstanding and collateral.
         delete positions[sponsor];
+
+        // Return fee-adjusted amount of collateral deleted from position.
+        return _getCollateral(remainingRawCollateral);
```

Comment:
Maybe I'm missing something, but shouldn't we compute the difference between the global collateral before and after the subtraction?

Reply:
in this case all of the position's collateral is being withdrawn from the global collateral, but I see a point in that there might rounding errors here. I'll adjust

Reply:
done